File: 242-valid-anagram/valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        ans1=sorted(s)
        ans2=sorted(t)
        if len(s)!=len(t):
            return False
        
        for i in range(len(s)):
            if ans1[i]!=ans2[i]:
                return False
        return True

File: 242-valid-anagram/valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        if(s.size()!=t.size()) return false;
        unordered_map<int,int>mp;
        for(int i=0;i<s.size();i++){
            mp[s[i]]++;
        }
        for(int i=0;i<t.size();i++){
            if(mp.find(t[i])==mp.end()){
                return false;
            }
            mp[t[i]]--;
        }
        for(auto it:mp){
            if(it.second!=0) return false;
        }
        return true;
    }
};